protocols/identify/src/protocol.rs

```diff
@@ -150,12 +150,14 @@ where
     }
 }
 
-pub(crate) async fn send<T>(io: T, info: Info) -> Result<(), UpgradeError>
+pub(crate) async fn send<T>(io: T, info: Info) -> Result<Info, UpgradeError>
 where
     T: AsyncWrite + Unpin,
 {
     trace!("Sending: {:?}", info);
 
+    let info_sent = info.clone();
```

Comment:
This might be a micro optimisation but if you change those two `into_iter` calls to just `iter()`, I think you can avoid cloning the `Info`!

That is because all of those calls in `.map` are already allocating anyway to produce the correct protobuf encoding, as such, I think we don't need to take ownership of `info` necessarily :)

Reply:
You might have to `.clone()` the `agent_version` and `protocol_version` specifically because those are strings.

Reply:
you're right. this minimizes the cloning.